homeassistant/components/light/mqtt.py

```diff
@@ -15,9 +15,14 @@
     ATTR_WHITE_VALUE, Light, SUPPORT_BRIGHTNESS, SUPPORT_COLOR_TEMP,
     SUPPORT_EFFECT, SUPPORT_COLOR, SUPPORT_WHITE_VALUE)
 from homeassistant.const import (
-    CONF_BRIGHTNESS, CONF_COLOR_TEMP, CONF_EFFECT, CONF_HS, CONF_NAME,
-    CONF_OPTIMISTIC, CONF_PAYLOAD_OFF, CONF_PAYLOAD_ON, STATE_ON,
-    CONF_RGB, CONF_STATE, CONF_VALUE_TEMPLATE, CONF_WHITE_VALUE, CONF_XY)
+    CONF_BRIGHTNESS, CONF_BRIGHTNESS_SCALE, CONF_COLOR_TEMP, CONF_EFFECT,
+    CONF_HS, CONF_NAME, CONF_OPTIMISTIC, CONF_PAYLOAD_OFF, CONF_PAYLOAD_ON,
+    STATE_ON, CONF_RGB, CONF_STATE, CONF_VALUE_TEMPLATE, CONF_WHITE_VALUE,
+    CONF_XY)
+from homeassistant.components.light.mqtt_json import (
+    PLATFORM_SCHEMA as PLATFORM_SCHEMA_JSON)
+from homeassistant.components.light.mqtt_json import (
+    _async_setup_entity as _async_setup_entity_json)
```

Comment:
Maybe consider using the `get_platform` import instead of this absolute one to support loading  [`custom_components`](https://developers.home-assistant.io/docs/en/creating_component_loading.html). I don't know how much this is enforced, but it would be best to call:

```python
# in async_discover
if discovery_payload['platform'] == 'mqtt_json':
    mqtt_json = get_platform(hass, 'light', 'mqtt_json')
    config = mqtt_json.PLATFORM_SCHEMA_JSON(discovery_payload)
    await mqtt_json._async_setup_entity_json(...)
```

See https://github.com/home-assistant/home-assistant/blob/ee696643cd7c63bf0cc88f5111c64b6e2d836640/homeassistant/loader.py#L8-L11

Reply:
Great, thanks!